### ISSUE TITLE
Issue with more than one h2o3 replica

### ISSUE THREAD

**Issue by fjudith** [h2oai/h2o-kubeflow#10]
Hello,

I'm getting the following error when i try to apply the AutoML tutorial.
I suspect it is all about client session stickyness to h2o3 instances, because it does not happen when i'm using only one replica.

![image](https://user-images.githubusercontent.com/17888379/38944981-8b6594c2-4335-11e8-9b33-afe4e2ae7a7a.png)

**Comment by fjudith** [comment]
```text
---------------------------------------------------------------------------
H2OResponseError                          Traceback (most recent call last)
<ipython-input-1-607126a22686> in <module>()
      6 test_loc = 'https://storage.googleapis.com/h2o-bq-large-dataset/higgs_test_5k.csv'
      7 
----> 8 train = h2o.import_file(train_loc)
      9 test = h2o.import_file(test_loc)

/opt/conda/lib/python3.6/site-packages/h2o/h2o.py in import_file(path, destination_frame, parse, header, sep, col_names, col_types, na_strings, pattern)
    412         return lazy_import(path, pattern)
    413     else:
--> 414         return H2OFrame()._import_parse(path, pattern, destination_frame, header, sep, col_names, col_types, na_strings)
    415 
    416 

/opt/conda/lib/python3.6/site-packages/h2o/frame.py in _import_parse(self, path, pattern, destination_frame, header, separator, column_names, column_types, na_strings)
    310             path = os.path.abspath(path)
    311         rawkey = h2o.lazy_import(path, pattern)
--> 312         self._parse(rawkey, destination_frame, header, separator, column_names, column_types, na_strings)
    313         return self
    314 

/opt/conda/lib/python3.6/site-packages/h2o/frame.py in _parse(self, rawkey, destination_frame, header, separator, column_names, column_types, na_strings)
    324                na_strings=None):
    325         setup = h2o.parse_setup(rawkey, destination_frame, header, separator, column_names, column_types, na_strings)
--> 326         return self._parse_raw(setup)
    327 
    328 

/opt/conda/lib/python3.6/site-packages/h2o/frame.py in _parse_raw(self, setup)
    349         p['source_frames'] = [_quoted(src['name']) for src in setup['source_frames']]
    350 
--> 351         H2OJob(h2o.api("POST /3/Parse", data=p), "Parse").poll()
    352         # Need to return a Frame here for nearly all callers
    353         # ... but job stats returns only a dest_key, requiring another REST call to get nrow/ncol

/opt/conda/lib/python3.6/site-packages/h2o/h2o.py in api(endpoint, data, json, filename, save_to)
    101     # type checks are performed in H2OConnection class
    102     _check_connection()
--> 103     return h2oconn.request(endpoint, data=data, json=json, filename=filename, save_to=save_to)
    104 
    105 

/opt/conda/lib/python3.6/site-packages/h2o/backend/connection.py in request(self, endpoint, data, json, filename, save_to)
    400                                     auth=self._auth, verify=self._verify_ssl_cert, proxies=self._proxies)
    401             self._log_end_transaction(start_time, resp)
--> 402             return self._process_response(resp, save_to)
    403 
    404         except (requests.exceptions.ConnectionError, requests.exceptions.HTTPError) as e:

/opt/conda/lib/python3.6/site-packages/h2o/backend/connection.py in _process_response(response, save_to)
    723         # Client errors (400 = "Bad Request", 404 = "Not Found", 412 = "Precondition Failed")
    724         if status_code in {400, 404, 412} and isinstance(data, (H2OErrorV3, H2OModelBuilderErrorV3)):
--> 725             raise H2OResponseError(data)
    726 
    727         # Server errors (notably 500 = "Server Error")

H2OResponseError: Server error water.exceptions.H2OIllegalArgumentException:
  Error: Missing data
  Request: POST /3/Parse
    data: {'destination_frame': 'higgs_train_10k.hex', 'parse_type': 'CSV', 'separator': '44', 'single_quotes': 'False', 'check_header': '1', 'number_columns': '29', 'chunk_size': '4194304', 'delete_on_done': 'True', 'blocking': 'False', 'column_types': '["Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric","Numeric"]', 'column_names': '["response","x1","x2","x3","x4","x5","x6","x7","x8","x9","x10","x11","x12","x13","x14","x15","x16","x17","x18","x19","x20","x21","x22","x23","x24","x25","x26","x27","x28"]', 'source_frames': '["https://storage.googleapis.com/h2o-bq-large-dataset/higgs_train_10k.csv"]'}
```

**Comment by fjudith** [comment]
Works by enforcing session affinity to `ClientIP`

![image](https://user-images.githubusercontent.com/17888379/38950962-61db7790-4347-11e8-8cba-0d7be5374bf8.png)


![image](https://user-images.githubusercontent.com/17888379/38950891-31023776-4347-11e8-9ae7-18a173120c92.png)

**Comment by fjudith** [comment]
## Backward test result



![image](https://user-images.githubusercontent.com/17888379/38951585-704656ae-4349-11e8-99b9-09fadbd0d32a.png)

**Comment by fjudith** [comment]
Committed the fix in PR #8

**Comment by nkpng2k** [comment]
@fjudith. Good find.

Merging fix with PR - Rebased on jupyter/tensorflow-notebook #8
closing for now
